lib/floe/workflow/choice_rule/data.rb

```diff
@@ -4,19 +4,35 @@ module Floe
   class Workflow
     class ChoiceRule
       class Data < Floe::Workflow::ChoiceRule
+        TYPES      = {"String" => :is_string?, "Numeric" => :is_numeric?, "Boolean" => :is_boolean?, "Timestamp" => :is_timestamp?, "Present" => :is_present?, "Null" => :is_null?}.freeze
+        COMPARES   = ["Equals", "LessThan", "GreaterThan", "LessThanEquals", "GreaterThanEquals", "Matches"].freeze
+        # e.g.: (Is)(String), (Is)(Present)
+        TYPE_CHECK = /^(Is)(#{TYPES.keys.join("|")})$/.freeze
+        # e.g.: (String)(LessThan)(Path), (Numeric)(GreaterThanEquals)()
+        OPERATION  = /^(#{(TYPES.keys - %w[Null Present]).join("|")})(#{COMPARES.join("|")})(Path)?$/.freeze
+
+        attr_reader :variable, :compare_key, :type, :value, :path
+
+        def initialize(_workflow, _name, payload)
+          super
+
+          @variable = parse_path("Variable", payload)
+          parse_compare_key
+          @value = parse_compare_value
+        end
+
         def true?(context, input)
+          return presence_check(context, input) if compare_key == "IsPresent"
+
           lhs = variable_value(context, input)
           rhs = compare_value(context, input)
 
-          validate!(lhs)
-
```

Comment:
Before:
- fetches a value, and verifies what ever came out is not null

After:
- fetches a value and verifies it existed in the input hash